### PR TITLE
Add vehicle icons to segmented radio buttons

### LINKED ISSUE #88: Add vehicle icons to segmented radio buttons
Segmented radio buttons should be able to deal with optional icons.

[AXA Brand Hub 2018-04-30 125246.zip](https://github.com/axa-ch/aletheia/files/1960276/AXA.Brand.Hub.2018-04-30.125246.zip)

### PR DESCRIPTION
**Pull request by mjenny**
Fixes #88

### CODE REVIEW

**Comment by mjenny** [comment]
@AndyOGo 
I had to leave preSelected inside JSON schema. Changing it to selected breaks functionality.

Still pending is the change for the icons. I committed the icons with motorcycle as a placeholder for others but did not yet implement the changes to actually display the icon.

I leave it up to you. ;)

**Comment by mjenny** [comment]
The difference between the Bullet Radio Buttons and the Segmented Radio Buttons is that we wrap the "selected" property inside an additional object "properties".
This seems to be unnecessary and complicates things:

BulletRadioButtons
```json
{
    name: 'radiosHorizontal',
    component: 'BulletRadioButtons',
    properties: {
      label: 'Anrede',
      options: [
        {
          label: 'Herr',
          value: '[BPM_VALUE132]',
          selected: false,
        },
        {
          label: 'Frau',
          value: '[BPM_VALUE232]',
          selected: true,
        },
      ],
      style: {},
    },
    actions: [
      {
        event: 'onChange',
        skipValidation: true,
      },
    ],
  }
```

SegmentedRadioButton
```json
{
    name: 'segmentVerticalVehicle',
    component: 'SegmentedRadioButtons',
    label: 'Was für ein fremdes Fahrzeug wurde beschädigt?',
    style: {
      orientation: 'vertical',
    },
    properties: {
      items: [
        {
          component: 'SegmentedRadioButtonItem',
          properties:
            {
              value: 'abc',
              selected: false,
              title: 'Auto',
              icon: 'car',
            },
        },
        {
          component: 'SegmentedRadioButtonItem',
          properties:
            {
              value: 'abc2',
              selected: true,
              title: 'Lieferwagen',
              icon: 'goodsTransport',
            },
        },
        {
          component: 'SegmentedRadioButtonItem',
          properties:
            {
              value: 'abc3',
              selected: false,
              title: 'Anderes Fahrzeug',
              subtitle: '(z.B. Lastwagen, Motorrad, Velo, ...)',
              icon: 'otherVehicles'
            },
        },
      ],
    },
    actions: [
      {
        event: 'onChange',
        skipValidation: true,
      },
    ],
  }
```
I guess we should change this.

What do you think?

@AndyOGo 
Could you make the changes to the JSON schema on confluence tomorrow if we need to (in case you are working on it)?

I commited the changes but feel free to revert them if necessary.

**Comment by AndyOGo** [comment]
I just realised we have a `<label>` inside a `<label>` here...

**Comment by AndyOGo** [comment]
@mjenny 
I have restored the `properties` group, just to be consistent with all other component schemas

File: src/components/segmented-radio-buttons/segmented-control/segmented-radio-button-item.scss [hunk 1]
```diff
@@ -45,3 +47,32 @@
     border-bottom-color: $color-prim-blue-azure;
   }
 }
+
+.a-segmented-radio-button-item__text--icon {
+  position: relative;
+  padding-left: $segmented-radio-button-item-x-padding + $segmented-radio-button-item__icon-size;
+  text-align: left;
+
+  .a-icon--wrapper {
```
Comment (AndyOGo) [inline src/components/segmented-radio-buttons/segmented-control/segmented-radio-button-item.scss]: This should be an element class, tracked in #92

File: src/components/segmented-radio-buttons/segmented-control/segmented-radio-button-item.scss [hunk 2]
```diff
@@ -45,3 +47,32 @@
...
+  position: absolute;
+
+  top: 50%;
+  left: $segmented-radio-button-item-x-padding / 2;
+  margin-top: $segmented-radio-button-item__icon-size / -2;
+  width: $segmented-radio-button-item__icon-size;
+  height: $segmented-radio-button-item__icon-size;
+}
+
+.a-segmented-radio-button-item__title {
```
Comment (AndyOGo) [inline src/components/segmented-radio-buttons/segmented-control/segmented-radio-button-item.scss]: Not sure if here are any specific design, typo changes required?

File: src/components/segmented-radio-buttons/segmented-control/segmented-radio-button-item.scss [hunk 3]
```diff
@@ -1,12 +1,14 @@
 @import '../../../../node_modules/@axa-ch/patterns-library/src/styles/mixins/typo';
 @import "../../../../node_modules/@axa-ch/patterns-library/src/styles/settings/colors";
 
+$segmented-radio-button-item-x-padding: 20px;
+$segmented-radio-button-item__icon-size: 40px;
```
Comment (AndyOGo) [inline src/components/segmented-radio-buttons/segmented-control/segmented-radio-button-item.scss]: @mjenny 
Not sure if this is the desired icon size

File: src/components/segmented-radio-buttons/segmented-control/segmented-radio-button-item.scss (hunk 2 above)
Comment (mjenny) [reply src/components/segmented-radio-buttons/segmented-control/segmented-radio-button-item.scss]: We will ask Luke on what exactly should be the icon size.

**Pull request merged**